lib/exercises.rb

```diff
@@ -1,19 +1,44 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n)
+# Space Complexity: o(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  return [] if strings.length == 0
+
+  anagrams = Hash.new([])
+  strings.each do |element|
+    sorted_letters = element.split('').sort
```

Comment:
You don't need to split the strings. 

```suggestion
    sorted_letters = element.sort
```